packages/field-base/src/field-mixin.js

```diff
@@ -369,12 +369,18 @@ export const FieldMixin = (superclass) =>
      * @protected
      */
     _invalidChanged(invalid) {
-      // Error message ID needs to be dynamically added / removed based on the validity
-      // Otherwise assistive technologies would announce the error, even if we hide it.
-      if (invalid) {
-        this._fieldAriaController.setErrorId(this._errorId);
-      } else {
-        this._fieldAriaController.setErrorId(null);
-      }
+      // This timeout is needed to prevent NVDA from announcing the error message twice:
+      // 1. Once adding the `[role=alert]` attribute by the `_updateErrorMessage` method (OK).
+      // 2. Once linking the error ID with the ARIA target here (unwanted).
+      // Related issue: https://github.com/vaadin/web-components/issues/3061.
+      setTimeout(() => {
```

Comment:
Let's use `Debouncer` to handle case when field becomes invalid and then immediately valid.

Reply:
I’m not sure if it is a real problem after all. The first callback will be postponed, but the possible second callback will be only executed after the first one because the event loop is a queue. 

The following test case passes even in the current implementation:

```js
it('should not contain error id after consecutive invalid/valid', async () => {
  element.invalid = true;
  element.invalid = false;
  await aTimeout(0);
  const aria = input.getAttribute('aria-labelledby');
  expect(aria).to.include(label.id);
  expect(aria).to.include(helper.id);
  expect(aria).to.not.include(error.id);
});
```